extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1225,50 +1223,73 @@ static public function getLocalEnvURL( $url, $forcedEnv = null ) {
 			// on fail at least return original url
 			return $url;
 		}
+
 		$protocol = $groups[1];
 		$server = $groups[2];
-		$address = $groups[3];
+		$path = $groups[3];
+
+		if ( !empty( $path ) ) {
+			$path = '/' . $path;
+		}
+
+		$localServer = self::getLocalEnvServer( $server, $forcedEnv );
+
+		if ( self::getCurrentEnvironment( $forcedEnv ) != WIKIA_ENV_PROD ) {
+			// Force HTTP in non-production environments
+			$protocol = 'http';
+		}
+
+		return $protocol . '://' . $localServer . $path;
+	}
 
-		if ( !empty( $address ) ) {
-			$address = '/' . $address;
+	public static function getCurrentEnvironment( $forcedEnv = false ) {
+		if ( $forcedEnv &&
+		     $forcedEnv !== WIKIA_ENV_DEV &&
+		     $forcedEnv !== WIKIA_ENV_SANDBOX ) {
+			return $forcedEnv;
+		} else {
+			return F::app()->wg->WikiaEnvironment;
 		}
+	}
 
+	public static function getLocalEnvServer( $server, $forcedEnv = null ) {
 		// strip env-specific pre- and suffixes for staging environment
 		$server = preg_replace( '/^(stable|preview|verify|sandbox-[a-z0-9]+)\./', '', $server );
-		if ( !empty( $wgDevDomain ) ) {
-			$server = str_replace( ".{$wgDevDomain}", '', $server );
+
```

Comment:
Replaced the line above with the regex again; this allows us to match more possible URLs that we get, translating them in to the proper local URL by the end of this method.